src/xcode/ENA/ENA/Source/Scenes/ContactDiary/StoreV2/ContactDiaryStoreV2.swift

```diff
@@ -0,0 +1,977 @@
+				return
+			}
+
+			result = .success(())
+		}
+
+		guard let _result = result else {
+			fatalError("[ContactDiaryStore] Result should not be nil.")
+		}
+
+		return _result
+	}
+
+	func removeAllLocations() -> DiaryStoringVoidResult {
+		return removeAllEntries(from: "Location")
+	}
+
+	func removeAllContactPersons() -> DiaryStoringVoidResult {
+		return removeAllEntries(from: "ContactPerson")
+	}
+
+	@discardableResult
+	func reset() -> DiaryStoringVoidResult {
+		let dropTablesResult = dropTables()
+		if case let .failure(error) = dropTablesResult {
+			return .failure(error)
+		}
+
+		let openAndSetupResult = openAndSetup()
+		if case .failure = openAndSetupResult {
+			return openAndSetupResult
+		}
+
+		var updateDiaryDaysResult: DiaryStoringVoidResult?
+		databaseQueue.inDatabase { database in
+			updateDiaryDaysResult = updateDiaryDays(with: database)
+		}
+		if case .failure(let error) = updateDiaryDaysResult {
+			return .failure(error)
+		}
+
+		return .success(())
+	}
+
+	func close() {
+		databaseQueue.close()
+	}
+
+	// MARK: - Private
+
+	private let dataRetentionPeriodInDays = 16 // Including today.
+	private let userVisiblePeriodInDays = 14 // Including today.
+	private let key: String
+	private let dateProvider: DateProviding
+	private let schema: ContactDiaryStoreSchemaV2
+
+	private var todayDateString: String {
+		dateFormatter.string(from: dateProvider.today)
+	}
+
+	private var dateFormatter: ISO8601DateFormatter = {
+		let dateFormatter = ISO8601DateFormatter.contactDiaryFormatter
+		return dateFormatter
+	}()
+
+	private var germanDateFormatter: DateFormatter = {
+		let dateFormatter = DateFormatter()
+		dateFormatter.dateStyle = .short
+		dateFormatter.timeStyle = .none
+		dateFormatter.locale = Locale(identifier: "de_DE")
+		return dateFormatter
+	}()
+
+	private let databaseQueue: FMDatabaseQueue
+	private let latestVersion: Int
+
+	private func openAndSetup() -> DiaryStoringVoidResult {
+		var errorResult: DiaryStoringVoidResult?
+		
+		databaseQueue.inDatabase { database in
+			Log.info("[ContactDiaryStore] Open and setup database.", log: .localData)
+			
+			let dbHandle = OpaquePointer(database.sqliteHandle)
+			guard CWASQLite.sqlite3_key(dbHandle, key, Int32(key.count)) == SQLITE_OK else {
+				Log.error("[ContactDiaryStore] Unable to set Key for encryption.", log: .localData)
+				errorResult = .failure(dbError(from: database))
+				return
+			}
+			
+			guard database.open() else {
+				Log.error("[ContactDiaryStore] Database could not be opened", log: .localData)
+				errorResult = .failure(dbError(from: database))
+				return
+			}
+			// We need to check for the following:
+			// (userVersion = 0) means: no database was created and stored previously by the user --> this is the first database --> create new updated schema
+			// if a database was stored but not the latest version --> migrate the old schema
+			if database.userVersion != 0 && database.userVersion < latestVersion {
```

Comment:
I think this is not needed if you place the migration after the schema creation.

Reply:
Turns out we need to place the migration before the schema creation, the schema will change the database version, so if we have an old database and we initialize the store --. > at the schema creation step we will change the database version ---> now if we try to migrate the migrator won't execute as the userDatabase version is now the latest because of the scema step